src/common/ical.rs

```diff
@@ -0,0 +1,142 @@
+use icalendar::{Calendar, Component};
+
+use crate::handlers::error::ResourceError;
+
+pub async fn parse_ical_uri(
```

Comment:
I have the feeling that this name is not right for what this function actually does. Maybe something more along the lines of `resolve_ical_by_uri`?

Reply:
How about `fetch_and_parse_ical_events`?